lib/engine_socketio.js

```diff
@@ -118,10 +118,10 @@ SocketIoEngine.prototype.step = function (requestSpec, ee) {
     return engineUtil.createLoopWithCount(requestSpec.count || -1, steps);
   }
 
-  let f = function(context, callback) {
+  let f = function(context, callback, socketio) {
     // Only process emit requests; delegate the rest to the HTTP engine (or think utility)
     if (requestSpec.think) {
-      return engineUtil.createThink(requestSpec, _.get(self.config, 'defaults.think', {}));
```

Comment:
We need to preserve the second argument to `createThink`, otherwise default think settings won't be applied.